### DISCUSSION TITLE
Github Actions fails at "brew audit --tap=homebrew/core"

### DISCUSSION THREAD

**Discussion by mchoo7** [Homebrew/discussions#3903] [Writing Formulae/Casks] [answered]
[PR](https://github.com/Homebrew/homebrew-core/pull/114641)
[log](https://github.com/Homebrew/homebrew-core/actions/runs/3441418824/jobs/5740932295)

During CI, `brew audit --tap=homebrew/core` is fails with following error message:

`alexjs:
    alexjs contains conflicting version recursive dependencies:
        openssl@3, openssl@1.1
      View these with brew deps --tree alexjs.`

How to fix this? Should I apply 'revision 1' for all those packages?

**Comment by carlocab** [comment, marked as answer]
This means that `alexjs` (recursively) depends on both `openssl@1.1` and `openssl@3` on versions of macOS prior to Catalina. Fixing this requires changing `alexjs` and/or its dependencies so that they all (recursively) depend on only `openssl@1.1` or only `openssl@3`.

Here's the output of `brew deps --tree alexjs` from your PR:
```
alexjs
└── node
    ├── brotli
    ├── c-ares
    ├── icu4c
    ├── libnghttp2
    ├── libuv
    ├── openssl@3
    │   └── ca-certificates
    ├── python
    │   ├── gdbm
    │   ├── mpdecimal
    │   ├── openssl@1.1
    │   │   └── ca-certificates
    │   ├── readline
    │   │   └── ncurses
    │   ├── sqlite
    │   │   ├── readline
    │   │   │   └── ncurses
    │   │   └── zlib
    │   ├── xz
    │   ├── bzip2
    │   ├── expat
    │   ├── libffi
    │   ├── libxcrypt
    │   ├── ncurses
    │   ├── unzip
    │   │   └── bzip2
    │   ├── zlib
    │   └── libnsl
    │       └── libtirpc
    │           └── krb5
    │               ├── openssl@1.1
    │               │   └── ca-certificates
    │               └── bison
    │                   └── m4
    └── zlib
```

**Reply by mchoo7** [reply]
However, I found that there are some formulas that have both python@3.10 and python@3.11. How can these formula pass the test?

**Reply by SMillerDev** [reply]
Because python is a scripted language that is often invoked at runtime and OpenSSL is a library that is linked at compile time. So while transient python version references are rare, OpenSSL ones are almost guaranteed.

**Reply by mchoo7** [reply]
Thanks for your answer!
